quic/s2n-quic-core/src/crypto/tls.rs

```diff
@@ -53,7 +51,12 @@ pub trait Context<Crypto: CryptoSuite> {
 
     fn on_server_name(
         &mut self,
-        server_name: Option<crate::application::ServerName>,
+        server_name: crate::application::ServerName,
+    ) -> Result<(), transport::Error>;
+
+    fn on_application_protocol(
+        &mut self,
+        application_protocol: &[u8],
```

Comment:
While you're here, this should be a `Bytes` instead so the TLS impl can also intern it.